crates/bevy_ui/src/ui_node.rs

```diff
@@ -2827,3 +2792,346 @@ impl Default for TextShadow {
         }
     }
 }
+
+/// A color stop for a gradient
+#[derive(Debug, Copy, Clone, PartialEq, Reflect)]
+#[reflect(Default, PartialEq, Debug)]
+#[cfg_attr(
+    feature = "serialize",
+    derive(serde::Serialize, serde::Deserialize),
+    reflect(Serialize, Deserialize)
+)]
+pub struct ColorStop {
```

Comment:
is this file supposed to have every single possible ui node component? might make sense to split it up a bit at some point cus 3k lines is pretty beefy

Reply:
Yep it's huge. Maybe a natural split would be to have all the required-for-every-node components in the `uinode` module and then make a second module called `components` (or something) for the optional components. I didn't add an extra module with this PR as it's huge enough already.